### PR TITLE
ETL: make retry-after behavior configurable

### PR DESCRIPTION
**Pull request by kislyuk**

### CODE REVIEW

File: dcplib/networking.py [hunk 1]
```diff
@@ -12,8 +12,9 @@
 logger = logging.getLogger(__name__)
 
 class Session(requests.Session):
+    obey_retry_after = True
     def resolve_redirects(self, resp, req, **kwargs):
-        if self.get_redirect_target(resp) and "Retry-After" in resp.headers:
+        if self.obey_retry_after and self.get_redirect_target(resp) and "Retry-After" in resp.headers:
```
Comment (mweiden) [inline dcplib/networking.py]: From our discussion: summarize how you're overriding urllib3 and requests to obey the Retry-After header

File: tests/test_etl.py [hunk 2]
```diff
@@ -90,7 +88,8 @@ def test_etl(self):
                 staging_directory=td,
                 content_type_patterns=["application/json"],
                 filename_patterns=["*.json"],
-                dss_client=MockDSSClient()
+                dss_client=MockDSSClient(),
+                http_client=MockDSSClient()
```
Comment (mweiden) [inline tests/test_etl.py]: nit: Separate out the `MockDSSClient` into a parent `MockHTTPClient` class and `MockDSSClient` child class for clarity

Comment (kislyuk) [reply tests/test_etl.py]: Done.

File: dcplib/networking.py (hunk 1 above)
Comment (kislyuk) [reply dcplib/networking.py]: Done.

**Comment by kislyuk** [comment]
Retry-after tests are already in place at https://github.com/HumanCellAtlas/dcplib/blob/master/tests/test_networking.py#L56.

Filed https://github.com/HumanCellAtlas/dcplib/issues/38 to track addition of more tests.

**Pull request merged**